### PR TITLE
blog: SigNoz MCP for log and trace investigation

### PR DESCRIPTION
**Pull request by gkarthi-signoz**
[BLOG->](https://signoz-web-git-mcp-blog-4-signoz.vercel.app/blog/signoz-mcp-log-trace-investigation/)
## Summary
Adds a new blog post: "Using SigNoz MCP for Log and Trace Investigation" — covering three use cases for natural language log and trace investigation: exploring logs without writing queries, reconstructing a bug from a trace ID, and tracing a failing request end-to-end across multiple services using the SigNoz MCP server.

## Motivation / Problem
Log and trace investigation traditionally requires knowing which service to filter by, which fields to query, and the specific time window — before any actual analysis can begin. This post shows how the SigNoz MCP server lets you start from a plain English symptom, trace ID, or failing endpoint and get to relevant signals immediately, without manual navigation.

## Changes
- Added `data/blog/signoz-mcp-log-trace-investigation.mdx` — blog post covering three MCP use cases: exploring logs without queries, reconstructing bugs from trace IDs, and tracing failing requests across services
- Added `public/img/blog/2026/05/mcp-blog-cover-4.webp` — cover image for the post
- Added `public/img/blog/2026/05/mcp-blog-section-cover-4.webp` — section visual for the post

## Type of Change

- [ ] **Docs** – Changes to `data/docs/**`
- [x] **Blog** – Changes to `data/blog/**`
- [ ] **Site Code** – Changes to `app/**`, `components/**`, `hooks/**`, `utils/**`, config, etc.
- [ ] **Redirects** – Renamed/moved docs or updated `next.config.js` redirects
- [ ] **Dependencies / CI / Scripts**
- [ ] **Other** – Explain below

## Impact

- [x] Documentation only
- [ ] UI changes
- [ ] Developer workflow
- [ ] Performance impact
- [ ] Breaking change

If breaking change, describe migration steps:

### General
- [ ] Branch is up to date with `main`
- [x] PR title follows conventional format (`type: description`)
- [ ] Self-reviewed the code
- [ ] Comments added for complex logic

### For all changes
- [ ] Built locally (`yarn build`) with no errors
- [ ] Ran `yarn lint` and fixed any issues
- [ ] Pre-commit hooks passed (or ran `yarn check:doc-redirects` / `yarn check:docs-metadata` if applicable)

### For docs changes (`data/docs/**`)
- [ ] Followed the docs author checklist in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [ ] Added/updated the page in `constants/docsSideNav.ts` if adding or moving a doc
- [ ] Any added docs images use WebP format and live under `public/img/docs/<topic>/`

### For blog changes
- [x] Followed the blog workflow in [contributing/blog-workflow.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/blog-workflow.md)
- [x] Frontmatter includes `title`, `date`, `author`, `tags` (and `canonicalUrl` if applicable)
- [x] Images use WebP format and live under `public/img/blog/<YYYY-MM>/`

### For site code changes
- [ ] Followed the site code playbook in [contributing/site-code.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/site-code.md)
- [ ] New dependencies are justified in the PR description (if any)

### For renamed or moved docs
- [ ] Followed the redirects and discovery section in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [ ] Added permanent redirect in `next.config.js` under `async redirects()`
- [ ] Updated internal links and sidebar in `constants/docsSideNav.ts`
- [ ] Ran `yarn check:doc-redirects` to verify

## Testing

- [ ] Tested locally
- [ ] Edge cases considered
- [ ] Existing functionality verified
- [ ] New tests added (if applicable)

Steps to test:
1. Navigate to `/blog/signoz-mcp-log-trace-investigation` and confirm the post renders correctly
2. Verify the cover image loads at `/img/blog/2026/05/mcp-blog-cover-4.webp`
3. Verify the section image loads at `/img/blog/2026/05/mcp-blog-section-cover-4.webp`
4. Confirm the post appears in the blog listing page

## Additional Context

---

**Note:** Submit as Draft by default. Mark "Ready for review" when checks pass and content is ready.

### CODE REVIEW

**Comment by gkarthi-signoz** [comment]
@claude /review

File: data/blog/signoz-mcp-log-trace-investigation.mdx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+<figure data-zoomable align='center'>
+    <img
+        src="/img/blog/2026/05/mcp-blog-cover-4.webp"
+        alt="Blog Cover"
+    />
+</figure>
+
+SigNoz makes this easier than most tools. Traces, logs, and metrics are correlated out of the box via OpenTelemetry, so you're not jumping between disconnected systems to piece together what happened. But even with that correlation in place, you still need to know which service to open, which filters to apply, and what to look for. The interface is powerful, but it still requires you to drive.
+
+The SigNoz MCP changes that dynamic. Instead of navigating to the right service and constructing the right filters, you describe what you're looking for in plain English and the MCP does the navigation for you. You start from a symptom, a trace ID, or a failing endpoint and it finds the relevant logs, traces, and correlated signals automatically.
```
[the remark this answers is not in the document]
Comment (gkarthi-signoz) [reply data/blog/signoz-mcp-log-trace-investigation.mdx]: no this is fine

File: data/blog/signoz-mcp-log-trace-investigation.mdx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+Each prompt builds on the last. You follow the signal rather than constructing the full query upfront. The MCP handles the service discovery, the filtering, and the aggregation and all you have to do is keep asking the next logical question until you have a root cause.
+
+- [Step-by-step guide](https://signoz.io/docs/ai/use-cases/natural-language-log-exploration/)
+
+## Reconstructing a Bug from a Trace ID
+
+- Support tickets and error trackers almost always come with a trace ID.
+- An error page shows a request ID.
+- A customer pastes a reference number from a failed transaction.
+
```
[the remark this answers is not in the document]
Comment (gkarthi-signoz) [reply data/blog/signoz-mcp-log-trace-investigation.mdx]: not needed in this case

**Comment by gkarthi-signoz** [comment]
@ankit01-oss @revmag plz review

File: data/blog/signoz-mcp-log-trace-investigation.mdx [hunk 3]
```diff
@@ -0,0 +1,187 @@
+---
+title: "Using SigNoz MCP for Log and Trace Investigation"
+slug: signoz-mcp-log-trace-investigation
```
Comment (LuffySama-Dev) [inline data/blog/signoz-mcp-log-trace-investigation.mdx]: Slug is not required meta section. It is removed after the CMS migrations.

File: data/blog/signoz-mcp-log-trace-investigation.mdx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+date: 2026-05-01
+tags: [SigNoz, MCP, Observability, Logs, Traces]
```
Comment (LuffySama-Dev) [inline data/blog/signoz-mcp-log-trace-investigation.mdx]: SEO best practices states to have two tags maximum.
One can be article/docs specific and other broader topic.

File: data/blog/signoz-mcp-log-trace-investigation.mdx [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+authors: [goutham_karthi]
+image: /img/blog/2026/05/mcp-blog-cover-4.webp
+description: Learn how the SigNoz MCP server lets you investigate logs and traces from plain English — whether you're starting from a vague symptom or a known failure.
+keywords: [SigNoz, MCP, observability, logs, traces, investigation, debugging, natural language, OpenTelemetry]
+---
```
Comment (LuffySama-Dev) [inline data/blog/signoz-mcp-log-trace-investigation.mdx:9]: To avoid creating orphan articles, it is recommended to have minimum three related articles added under the related articles list in front matter.

Example Format
related_articles:
  - '/blog/launch-week-1'
  - '/blog/launch-week-4-recap'
  - '/blog/open-source-observability'
  - '/blog/launching-signoz-single-binary'

File: data/blog/signoz-mcp-log-trace-investigation.mdx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+Each prompt builds on the last. You follow the signal rather than constructing the full query upfront. The MCP handles:
+
+- **service discovery**
+- **filtering**
+- **aggregation**
+
+All you have to do is keep asking the next logical question until you have a root cause.
+
+[Step-by-step guide →](https://signoz.io/docs/ai/use-cases/natural-language-log-exploration/)
```
Comment (LuffySama-Dev) [inline data/blog/signoz-mcp-log-trace-investigation.mdx]: When interlinking like this we try to use keyword rather than using here, this guide, step by step guide. 

You check out our guide on [Natural Language Log Exploration](https://signoz.io/docs/ai/use-cases/natural-language-log-exploration/) for more details ... etc

**Review by LuffySama-Dev — CHANGES_REQUESTED** [review]

File: data/blog/signoz-mcp-log-trace-investigation.mdx [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+## The Problem with Traditional Log and Trace Investigation
+
+<figure data-zoomable align='center'>
+    <img
+        src="/img/blog/2026/05/mcp-blog-section-cover-4.webp"
+        alt="Section Cover"
+    />
+</figure>
+
+The challenge with log and trace investigation isn't that the data isn't there. It's that accessing it requires context you often don't have right at the start of an investigation.
```
Comment (mahakkaur22) [inline data/blog/signoz-mcp-log-trace-investigation.mdx]: This section repeats some of the setup already covered in the intro: logs/traces have the data, but investigations are hard because the reader lacks starting context. To keep the flow moving, can we compress the first two paragraphs under this H2 and get to the examples faster?

Suggested update: replace these two lines:

> “The challenge with log and trace investigation isn't that the data isn't there. It's that accessing it requires context you often don't have right at the start of an investigation.”


> “When an incident starts or a bug gets reported, the information you typically have is vague.”


With:

> The hard part of log and trace investigation is usually not missing data. It is missing starting context: the service name, the right fields, the time window, or the trace ID. In practice, that starting point often looks vague:

(then the existing bullets follow)

- **Slack message**: "search results are stale"
- **Support ticket**: a user ID and a timestamp
- **Endpoint**: returning 500s

File: data/blog/signoz-mcp-log-trace-investigation.mdx (hunk 5 above)
Comment (gkarthi-signoz) [reply data/blog/signoz-mcp-log-trace-investigation.mdx:9]: will make this change after are all blogs are merged so i can cross reference.

**Review by LuffySama-Dev — APPROVED** [review]

**Review by LuffySama-Dev — APPROVED** [review]
